### ISSUE TITLE
Set `query_id` in request

### ISSUE THREAD

**Issue by ibradwan** [DarkWanderer/ClickHouse.Client#79]
Hi there,

I would like to know if there's a way to pass HTTP Get parameters in order to include them in the ClickHouse request.

What we want to do is to pass the `query_id` parameter in order to set the id of the query.

**Comment by DarkWanderer** [comment]
Hi,

You can use `ClickHouseConnection.CustomSettings` property to pass arbitrary URI parameters - including `query_id`. The only catch is those parameters are tied to connection, not to command, but it should work for most cases:
```
using var connection = new ClickHouseConnection(...);
connection.CustomSettings.Add("replace_running_query", 1);
connection.CustomSettings.Add("query_id", "my_query");
```

**Comment by ibradwan** [comment]
Thanks, but I suppose creating a connection per query (10s per second) could have some side effects, is this right?

**Comment by DarkWanderer** [comment]
You can use same connection and change `query_id` value between queries, if you can make sure to use it in one thread at a time

What is your use case - why do you need `query_id` if not for `replace_running_query`?

**Comment by ibradwan** [comment]
> Yes, creating multuple connections very fast can lead to HTTP client pool exhaustion and also will lead to tons of extra queries. You can use same connection and change `query_id` value between queries, if you can make sure to use it in one thread at a time

Yeah this is pretty limiting, esp that CH queries could take time, so running them serially is not a good option for us.

> What is your use case - why do you need `query_id` if not for `replace_running_query`?

We need to tie the `query_id` to another identifier on our application level (i.e., `request_id`) for debugability. Also, we need to know the `query_id` in advance to be able to `KILL` the query whenever the user drops the request to save some CH resources.

**Comment by DarkWanderer** [comment]
Ok, I'll think about adding it to `ClickHouseCommand` public API

**Comment by ibradwan** [comment]
Hi there,

I have tried to support this simple functionality locally and I think I have something that we can build on. Could you please give me access such that I can push and make a PR?

Thanks!

**Comment by DarkWanderer** [comment]
Sure, you can fork the repository and make PR from the fork

**Comment by OmarBazaraa** [comment]
Hi @DarkWanderer,
It will be extremely useful for us to support setting the `query_id` in advance. We need correlate different identifiers together, and most importantly, we need to be able to kill queries on timeout or on cancellation.

**Comment by DarkWanderer** [comment]
Implemented in #120 (sorry for long delay)

**Comment by DarkWanderer** [comment]
Released in [4.2.0](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/4.2.0)
